apps/docs/content/docs/editor.mdx

```diff
@@ -694,6 +694,36 @@ editor.run(
 )
 ```
 
+### Hiding shapes
+
+You can pass a `isShapeHidden` predicate when configuring tldraw. This predicate will be called for each shape to determine if it should be hidden.
+
+Hidden shapes will still be present in the store, but
+
+- They will not be shown on the canvas
+- They will not be included in hit test results via `editor.getShapeAtPoint` and `editor.getShapesAtPoint`
+
+Otherwise, they will behave as normal. If you need to restrict certain interactions with hidden shapes, e.g. preventing the user from having a hidden shape selected, you need to handle that in your own code.
```

Comment:
this is interesting to think through, I would assume `getShapeAtPoint` and `getShapesAtPoint` would stop the shape from being selectable, but I would assume that if you selected a parent that is visible it would move the inner shape despite it being hidden (this seems to be the case in the example so that's good!).

also would assume bindings would move with them, but i think that comes with the meal as bindings inherently work that way, but thought i'd just double check.

other than that, is this note basically stating that another user might *already* have the shape selected so it could feasibly still be moved etc on its own, and if we want to disable that we should handle it? or is this saying that by default shapes will be selectable despite being hidden (if so, via which method would that be so?).

Reply:
> is this note basically stating that another user might already have the shape selected so it could feasibly still be moved etc on its own, and if we want to disable that we should handle it?

Yes.

> or is this saying that by default shapes will be selectable despite being hidden

Also yes. I'll reword to make this clearer.

> (if so, via which method would that be so?)

It mainly depends on the app and how 'hiddenness' is used to implement features. One that applies to all apps would be the `selectAll` feature (cmd+a). This selects all the unlocked shapes (including the hidden ones) because that might be what you want if you're building an app with a layers panel. But you wouldn't want cmd+a to select hidden shapes in the context of a multiplayer app with 'private' shapes.

Reply:
sweet, I think I'll have a better idea on any more thoughts when implementing on our end.

generally I think most folks will expect a hidden shape to work how it does in figma, so the more that can be baked in the better, but trust y'all to think through the bigger picture as realise you've got more use cases to bear in mind.